src/DotNetWorker.Core/Context/FunctionContext.cs

```diff
@@ -59,5 +60,10 @@ public abstract class FunctionContext
         /// Gets a collection containing the features supported by this context.
         /// </summary>
         public abstract IInvocationFeatures Features { get; }
+
+        /// <summary>
+        /// Gets the <see cref="CancellationToken"/> that signals a function invocation is being cancelled.
+        /// </summary>
+        public virtual CancellationToken CancellationToken { get; }
```

Comment:
Wonder we should have this as a property of FunctionContext. Instead, what If we allow customers to just use a parameter of the type `CancellationToken`  in their functions? If they do not use this parameter, we do not need to keep an entry in the dictionary (which tracks invocation to CTS). That also makes the programming experience consistent with the [in-proc model](https://docs.microsoft.com/en-us/azure/azure-functions/functions-dotnet-class-library?tabs=v2%2Ccmd#cancellation-tokens)

Reply:
I hope this should not block your host changes though Lilian.

Reply:
I actually had this in mind when I first started out as I wanted it to be like the in-proc example, but I just didn't know how to expose a new parameter like this for the users - if this is something you know how to do, would you mind walking be through it? Or you can branch off this branch and make those changes, then we can walk through it together. 

Technically doesn't block host changes since I have validated that we are hitting the Cancellation handler, and how we deal with that in the worker is totally up to us and shouldn't impact the host. But it is probably a good idea to validate E2E with host if we make those changes

Reply:
Had a sync with Shyju and updated the code to use a CancellationToken parameter